09_json/willshen8/cmd/puppy-server/main.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/anz-bank/go-course/09_json/willshen8/pkg/puppy"
+	jsonify "github.com/anz-bank/go-course/09_json/willshen8/pkg/puppy/json"
+	store "github.com/anz-bank/go-course/09_json/willshen8/pkg/puppy/store"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var out io.Writer = os.Stdout
+
+var (
+	args     = os.Args[1:]
+	fileName = kingpin.Flag("data", "Puppy Json File Name").Short('d').String()
+)
+
+func main() {
+	processCommandLineArgs()
+}
+
+func processCommandLineArgs() {
+	var jsonData []byte
+	var decodedPuppies []puppy.Puppy
+	var mystore = store.NewMapStore()
+
+	if _, parseError := kingpin.CommandLine.Parse(args); parseError != nil {
+		panic(parseError)
+	}
+
+	jsonData, err := ioutil.ReadFile(*fileName)
+
+	if err != nil {
+		panic(err)
+	}
+
+	if unmarshalErr := jsonify.UnmarshalPuppy(jsonData, &decodedPuppies); unmarshalErr != nil {
```

Comment:
i'd call this one err to: `if err = jsonify....`

Reply:
ok